### ISSUE TITLE
`rust-analyzer` only provides code actions

### ISSUE THREAD

**Issue by noib3** [zed-industries/feedback#100]
**Describe the bug**

The Rust LSP seems to not be working at all apart from providing code actions. Everything else from code completion, mouse "hovers", go to definition, go to next/prev diagnostic is absent.  It pops up in `ps aux | grep rust-analyzer` so it should be running. Also, no error message is reported in the project diagnostics area.

**To Reproduce**

Open a Rust project -> start coding :)

**Expected behavior**

`rust-analyzer` prompts you with code completions as you type, and key bindings like `F8`, `Shift + F8`, `F12` all work.

**Environment:**
 - Architecture: Intel
 - Mac OS Version: `12.4`
 - Zed Version: `0.39.0`

**Comment by mikayla-maki** [comment]
I can't seem to recreate this bug on my end, is there anything in your status bar to the bottom left?

**Comment by mikayla-maki** [comment]
I can't seem to recreate this bug, do you have more information? My diagnostics pane, jump to definition, code completion, etc. are all running.

**Comment by noib3** [comment]
`~/.zed/rust-analyzer/rust-analyzer-2022-06-20 --version`'s output is `rust-analyzer 0.0.0 (427061da1 2022-06-19)`. Not sure what other info I can give you :/

The docs don't mention it, but if there's a log file saved somewhere I'd be happy to upload it.

**Comment by mikayla-maki** [comment]
I just checked with the team and there should be one at `~/Library/Logs/Zed/Zed.log`

**Comment by noib3** [comment]
[This](https://pastebin.com/AAuhBYcJ) is the log file that I got. Turns out it was just an issue caused by the non-standard way in which the Nix package manager handles dependency paths.

For anyone stumbling on the same issue, my solution was to install the [Rust overlay](https://github.com/oxalica/rust-overlay) with
```nix
    (pkgs.rust-bin.stable.latest.default.override {
      extensions = [ "rust-src" ];
    })
```
instead of just `pkgs.rust-bin.stable.latest.default`.
